Synview-backend/middleware/auth_middleware.ts

```diff
@@ -0,0 +1,85 @@
+import { create, verify, Context, decode } from "../deps.ts";
+import { z } from "zod";
+import getToken from "./jwt.ts";
+import { Session } from "../deps.ts"
+/// Token Auth
+let key: CryptoKey;
+type AppState = {
+    session: Session
+}
+
+export const UserPayload = z.object({
+  ["username"]: z.string(),
+  ["role"]: z.string(),
+  ["id"]: z.number(),
+});
+
+export async function createToken(payload: any): Promise<string> {
+  key = await crypto.subtle.generateKey(
+    { name: "HMAC", hash: "SHA-512" },
+    true,
+    ["sign", "verify"]
+  );
+
+  return create({ alg: "HS512", typ: "JWT" }, payload, key);
+}
+export function getPayload(body: any) {
+  try {
+    const parsedBody = UserPayload.parse(body);
+    return { ...parsedBody };
+  } catch (error) {
+    throw new Error("Couldnt parse User Payload" + error);
+  }
+}
+
+export default async function AuthMiddleware(
+  context: Context<AppState>,
+  next: () => Promise<unknown>
+) {
+  try {
+    const auth = context.state.session.get("Authorization")
+    console.log("this is authmiddleware")
+    console.log(auth)
```

Comment:
It is very rare to see `console.log` in production code. I would recommend creating an abstracted `Logger` type, with different warning levels, and in your `env` you can have a `logLevel` flag. this way, in production, all logging except error logging can be shut off